### PR TITLE
add step to login to ghcr

### PR DESCRIPTION
**Pull request by markgllin**
required if base images are sourced from ghcr

### CODE REVIEW

**Review by mkhinda29 — APPROVED** [review]

File: .github/workflows/publish-and-deploy.yml [hunk 1]
```diff
@@ -40,6 +45,9 @@ on:
       GITHUB_PAT:
         description: "A GitHub access token that has repo and read:packages scopes"
         required: false
+      GITHUB_USERNAME:
```
Comment (davegallant) [inline .github/workflows/publish-and-deploy.yml:48]: This shouldn't be needed

**Review by davegallant — CHANGES_REQUESTED** [review]
Username shouldn't be necessary?

File: .github/workflows/publish-and-deploy.yml (hunk 1 above)
Comment (davegallant) [reply .github/workflows/publish-and-deploy.yml:48]: https://sourcegraph.rewind.com/search?q=context:global+uses:+docker/login-action%40v1&patternType=literal

Comment (davegallant) [reply .github/workflows/publish-and-deploy.yml:48]: https://github.com/rewindio/membrane/blob/7a0cd424a5da12d59c07c1fad27b8822abaee563/.github/workflows/docker.yml#L29

File: .github/workflows/publish-and-deploy.yml [hunk 2]
```diff
@@ -30,6 +30,11 @@ on:
         description: "The target to build"
         required: false
         type: string
+      login_to_ghcr:
+        description: "Login to GHCR"
+        required: false
+        type: boolean
+        default: false
```
Comment (davegallant) [inline .github/workflows/publish-and-deploy.yml:37]: Should the default be true? Almost all workflows will likely require this?

**Review by davegallant — APPROVED** [review]

**Pull request merged**